### DISCUSSION TITLE
[mini.pick] How to end a cli process on `<C-C>`?

### DISCUSSION THREAD

**Discussion by jamestansx** [nvim-mini/mini.nvim#1070] [Q&A] [answered]
Hii, yet another question : )

Let's say I have a cli process that start from `builtin.cli`, and it took a long time to process, if I tried to stop the picker, the process would still be running in the background.

This is not ideal, as I might open file picker on my home directory that `fd` took too long to process, and I would like to cancel it. But now the process will still be running in the background and consume my cpu resources.

Perhaps not using `builtin.cli` and handle the process creation and call `start()` myself?

**Comment by echasnovski** [comment]
Hi!

I don't think there is a way to do that currently in `builtin.cli()`. But there probably should be. I'll think about it.

**Comment by echasnovski** [comment, marked as answer]
Stopping picker on latest `main` branch now also kills the process if it is still running. It means not only with `<C-c>`, but also any "regular" way to stop the picker (like `<Esc>`). Tested both with `builtin.files()` and `cli()` with `echo 20` command; both seem to work.

@jamestansx, thanks for the observation and suggestion!

**Reply by jamestansx** [reply]
Tested, it works! Thank you for this awesome plugin!
